.github/workflows/node-ci-push.yml

```diff
@@ -0,0 +1,120 @@
+name: Node CI (landed on master)
+
+on:
+  push:
+    branches:
+    - master
+
+# Our jobs run like this to minimize wasting resource cycles:
+#   1. Prime caches for primary configuration (ubuntu on node 14).
+#      This way the next two jobs can run in parallel but rely on this primed
+#      cache.
+#   2. Coverage
+#   3. Chromatic autoApprove on squash commits
+#
+# For pushes directly to a branch, we assume a PR has been used with wider
+# checks, this just makes sure our coverage data is up-to-date.
+jobs:
+  prime_cache_primary:
+    name: Prime node_modules cache for primary configuration
+    env:
+      CI: true
+    runs-on: ${{ matrix.os }}
+    strategy:
+      # Use a matrix as it means we get the version info in the job name
+      # which is very helpful.
+      matrix:
+        os: [ubuntu-latest]
+        node-version: [14.x]
+    steps:
+    - uses: actions/checkout@v2
+    - name: Use Node.js ${{ matrix.node-version }}
+      uses: actions/setup-node@v2
+      with:
+        node-version: ${{ matrix.node-version }}
+    - uses: actions/cache@v2
+      id: node-modules-cache # use this to check for `cache-hit` (`steps.node-modules-cache.outputs.cache-hit != 'true'`)
+      with:
+        path: node_modules
+        key: ${{ runner.os }}${{ matrix.node-version }}-node-modules-${{ hashFiles('**/yarn.lock') }}
+        restore-keys: |
+          ${{ runner.os }}${{ matrix.node-version }}-node-modules-
+    - name: Install dependencies
+      if: steps.node-modules-cache.outputs.cache-hit != 'true'
+      run: yarn install --frozen-lockfile
+
+
+  coverage:
+    needs: [prime_cache_primary]
+    name: Gather coverage
+    env:
+      CI: true
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        # Use a matrix as it means we get the version info in the job name
+        # which is very helpful.
+        os: [ubuntu-latest]
+        node-version: [14.x]
+    steps:
+    - uses: actions/checkout@v2
+    - name: Use Node.js ${{ matrix.node-version }}
+      uses: actions/setup-node@v2
+      with:
+        node-version: ${{ matrix.node-version }}
+    - uses: actions/cache@v2
+      id: node-modules-cache # use this to check for `cache-hit` (`steps.node-modules-cache.outputs.cache-hit != 'true'`)
+      with:
+        path: node_modules
+        key: ${{ runner.os }}${{ matrix.node-version }}-node-modules-${{ hashFiles('**/yarn.lock') }}
+        restore-keys: |
+          ${{ runner.os }}${{ matrix.node-version }}-node-modules-
+    - name: Install dependencies
+      if: steps.node-modules-cache.outputs.cache-hit != 'true'
+      run: yarn install --frozen-lockfile
```

Comment:
Is there any way we could bypass these steps since we have `needs: [prime_cache_primary]` at the start of this job?